Vagrantfile

```diff
@@ -15,29 +15,82 @@
 #   See the License for the specific language governing permissions and
 #   limitations under the License.
 
-# Vagrantfile for cgroup2
+# Vagrantfile for cgroup2 and SELinux
 Vagrant.configure("2") do |config|
   config.vm.box = "fedora/32-cloud-base"
   config.vm.provider :virtualbox do |v|
     v.memory = 2048
     v.cpus = 2
+    v.customize ["modifyvm", :id, "--audio", "none"]
   end
   config.vm.provider :libvirt do |v|
     v.memory = 2048
     v.cpus = 2
   end
-  config.vm.provision "shell", env: {"RUNC_FLAVOR"=>ENV["RUNC_FLAVOR"]}, inline: <<-SHELL
-    set -eux -o pipefail
-    # configuration
-    GO_VERSION="1.13.14"
 
-    # install dnf deps
-    dnf install -y container-selinux gcc git iptables libseccomp-devel lsof make
+  # Disabled by default. To run:
+  #   vagrant up --provision-with=upgrade-packages
+  # To upgrade only specific packages:
+  #   UPGRADE_PACKAGES=selinux vagrant up --provision-with=upgrade-packages
+  #
+  config.vm.provision "upgrade-packages", type: "shell", run: "never" do |sh|
+    sh.upload_path = "/tmp/vagrant-upgrade-packages"
+    sh.env = {
+        'UPGRADE_PACKAGES': ENV['UPGRADE_PACKAGES'],
+    }
+    sh.inline = <<~SHELL
+        #!/usr/bin/env bash
+        set -eux -o pipefail
+        dnf -y upgrade ${UPGRADE_PACKAGES}
+    SHELL
+  end
 
-    # install Go
-    curl -fsSL "https://dl.google.com/go/go${GO_VERSION}.linux-amd64.tar.gz" | tar Cxz /usr/local
+  # To re-run, installing CNI from RPM:
+  #   INSTALL_PACKAGES="containernetworking-plugins" vagrant up --provision-with=install-packages
+  #
+  config.vm.provision "install-packages", type: "shell", run: "once" do |sh|
+    sh.upload_path = "/tmp/vagrant-install-packages"
+    sh.env = {
+        'INSTALL_PACKAGES': ENV['INSTALL_PACKAGES'],
+    }
+    sh.inline = <<~SHELL
+        #!/usr/bin/env bash
+        set -eux -o pipefail
+        dnf -y install \
+            container-selinux \
+            curl \
+            gcc \
+            git \
+            iptables \
+            less \
+            libseccomp-devel \
+            libselinux-devel \
+            lsof \
+            make \
+            mercurial \
+            selinux-policy \
+            selinux-policy-devel \
+            selinux-policy-targeted \
+            socat \
+            subversion \
```

Comment:
Why?

Reply:
subversion is answered here: https://github.com/containerd/containerd/pull/4451#discussion_r465419887

the selinux-* make sure the latest of these are installed with the selinux-devel specifically needed to debug/develop container-selinux from source